app/lib/audit/backend.dart

```diff
@@ -136,4 +139,56 @@ class AuditBackend {
     InvalidInputException.check(parsed != null, 'Unable to parse `before`.');
     return parsed!;
   }
+
+  /// Returns the entries from the last day.
+  ///
+  /// Keeps the [_cachedRecords] fields updated, and lists only the entries
+  /// up to the last query.
+  Future<List<AuditLogRecord>> getEntriesFromLastDay() async {
+    final now = clock.now().toUtc();
+    final cachedAge = now.difference(_cachedRecords.timestamp);
+
+    // fast track for requests within 5 seconds:
+    if (cachedAge.inSeconds < 5) {
+      return _cachedRecords.records;
+    }
+
+    // calculate window to query
+    final day = const Duration(days: 1);
+    var window = cachedAge > day ? day : (day - cachedAge);
+    if (window < Duration(minutes: 2)) {
+      window = Duration(minutes: 2);
+    }
+
+    final query = dbService.query<AuditLogRecord>()
+      ..filter('created >', now.subtract(window));
+    final current = await query.run().toList();
+
+    // merge records from cache and current query
+    final recordMap = <String, AuditLogRecord>{};
+    for (final r in _cachedRecords.records) {
+      recordMap[r.id!] = r;
+    }
+    for (final r in current) {
+      recordMap[r.id!] = r;
+    }
+    // filter records that are within the last day
+    final records = recordMap.values
```

Comment:
We could also use `Set` instead of map... like:
```dart
final newIds = newRecords.map(r => r.id).toSet();
final records = [
  ...cachedRecords.records.where(r => !newIds.contains(r.id)),
  ...newRecords,
];
```
Or even as simple as:

```dart
final ids = <String>{};
final records = [
  ...cachedRecords.records,
  ...newRecords,
].where(r => ids.add(r.id!)).toList();
```

The ideal, would of course be something like:
```dart
extension UniqueList<T> on List<T> {
  List<T> uniqueBy<S>(S Function(T element) byKey) {
    final uniqueKeys = <S>{};
    return this.where(e => uniqueKeys.add(byKey(e))).toList();
  }
}

// then simply do:
final records = [
  ...cachedRecords.records,
  ...newRecords,
].uniqueBy(r => r.id!);
```

But downside of set/map is that we do allocations, and these are probably more expensive than just looping through the small list of newRecords many times..

Reply:
Updated to use the current id set as a filter.